lib/react.dart

```diff
@@ -5,14 +5,28 @@
 /// A Dart library for building UI using ReactJS.
 library react;
 
+import 'dart:html' show Element;
+
 /// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/top-level-api.html#react.component)
 /// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/component-api.html)
 abstract class Component {
+  Map _props;
+  Map _state = {};
+  dynamic _ref;
+
   /// ReactJS `Component` props.
-  Map props;
+  Map get props => _props;
+  set props(Map value) => _props = value;
```

Comment:
@greglittlefield-wf I just tried that, and I still get the following errors in over_react:

```
ERROR: Field declaration Component.ref cannot be overridden in UiComponent. ([over_react] lib/src/component_declaration/component_base.dart:106)
ERROR: Field declaration Component.props cannot be overridden in UiComponent. ([over_react] lib/src/component_declaration/component_base.dart:179)
ERROR: Field declaration Component.state cannot be overridden in UiStatefulComponent. ([over_react] lib/src/component_declaration/component_base.dart:230)
```